### PR TITLE
[MO] Generalized StridedSlice -> Squeeze/Unsqueeze

### PR DESCRIPTION
**Pull request by pavel-esir**
**Root cause analysis**:  For `NHWC` nets Squeezing\unsqueezing should be done in the original FW layout regardless of operation with which they are expressed Reshape, StridedSlice (SS), Squeeze, Unsqueeze. We have a transformation that replaces SS with shring\new_axis masks by Squeeze\Unsqueeze. And for Squeeze\Unsqueese there is a `reinterp_shape` attribute which takes care of inserting Transposes to restore original FW `NHWC` layout.

But it is triggered only if squeezing/unsqueezing is the only thing that SS does. That is not always the case, e.g. PixelLink has SS which slices data and at the same time Squeezes. 

**Solution**: Generalized solution from #6693. Squeeze/Unsqueeze are inserted even if StridedSlice does slicing as well. Also generalized for cases when SS has shrink_axis\new_axis masks at the same time.

Ticket: CVS-73336

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests" n/a
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list: n/a
* [x]  Guide on how to convert the **public** model: n/a
* [x]  User guide update: n/a

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

File: tools/mo/unit_tests/mo/middle/StridedSliceShrinkNewAxisReplacer_test.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            },
+            nodes_with_edges_only=True
+        )
+
+        ReplaceStridedSliceShrinkNewAxisWithSqueezeUnsqueeze().find_and_replace_pattern(graph)
+        graph = partial_infer(graph)
+        (flag, resp) = compare_graphs(graph, ref_graph, 'result', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+    def test_replace_shrink_new_axis_with_squeeze_unsqueeze(self):
```
Comment (sadolini) [inline tools/mo/unit_tests/mo/middle/StridedSliceShrinkNewAxisReplacer_test.py:111]: Please add test when new_axis index is less then shrink

**Review by sadolini — COMMENTED** [review]
Looks good to me. But could you check if new_axis/sharink_axis masks influence on begin/end masks: should we adapt it accordingly? I remember that we had very similar transformation before and have bug with removing new_axis_mask/shrink_axis and shape inference in IE. A lot of changes were done for SS and possibly now it will not be a problem, but testing should be done carefully.

File: tools/mo/unit_tests/mo/middle/StridedSliceShrinkNewAxisReplacer_test.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/unit_tests/mo/middle/StridedSliceShrinkNewAxisReplacer_test.py:111]: I started to check these cases, and you are rights. Removing `new_axis_mask` if it's less than other masks we have to do some extra modifications.

Currently I am trying to solve this. Will update when it's done

**Comment by pavel-esir** [comment]
Closing this PR. Decided to continue with the simpler solution #9622

**Pull request closed without merging**